src/librustc_resolve/resolve_imports.rs

```diff
@@ -52,7 +52,10 @@ pub enum ImportDirectiveSubclass<'a> {
         value_result: Cell<Result<&'a NameBinding<'a>, Determinacy>>,
         type_result: Cell<Result<&'a NameBinding<'a>, Determinacy>>,
     },
-    GlobImport { is_prelude: bool },
+    GlobImport {
+        is_prelude: bool,
+        max_vis: Cell<ty::Visibility>, // The visibility of the greatest reexport.
```

Comment:
Is this basically an optimisation to not even look at globs which would import nothing? Because otherwise, I would not expect the visibility to matter here.